duopull/duopull.go

```diff
@@ -0,0 +1,698 @@
+package duopull
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/mozilla-services/foxsec-pipeline-contrib/common"
+
+	"cloud.google.com/go/datastore"
+	stackdriver "cloud.google.com/go/logging"
+	log "github.com/sirupsen/logrus"
+	"go.mozilla.org/mozlogrus"
+)
+
+const (
+	_ = iota
+	debugOff
+	debugDuo
+	debugGCP
+)
+
+var debug = debugOff
+
+const (
+	ADMIN_ENDPOINT     = "/admin/v1/logs/administrator"
+	AUTH_ENDPOINT      = "/admin/v2/logs/authentication"
+	TELEPHONY_ENDPOINT = "/admin/v1/logs/telephony"
+
+	LOGGER_NAME = "duopull"
+
+	MINTIME_KIND      = "mintime"
+	MINTIME_KEY       = "mintime"
+	MINTIME_NAMESPACE = "mintime"
+)
+
+var (
+	PROJECT_ID string
+	KEYNAME    string
+
+	cfg config
+)
+
+func init() {
+	mozlogrus.Enable("duopull")
+	PROJECT_ID = os.Getenv("GCP_PROJECT")
+	KEYNAME = os.Getenv("KMS_KEYNAME")
```

Comment:
Are `GCP_PROJECT` and `KMS_KEYNAME` required variables? If so we should probably document that with the other required variables in the README file, and potentially check for these in `validate`?

Reply:
Yeah, they are required in certain settings. Added to docs and added checks.